external-crates/move/crates/move-compiler/tests/sui_mode/private_generics/internal.move

```diff
@@ -0,0 +1,40 @@
+// tests that `internal::permit` can only be called by the module that defines the type
+
+module a::m {
+  use a::other::NonInternalType;
+  use std::internal::{Self, Permit};
+
+  struct Internal {}
+
+  struct InternalGeneric<phantom T> {}
+
+  // success
+  public fun t1() {
+    internal::permit<Internal>();
+  }
+
+  // fail: non-internal type
+  public fun t2() {
+    internal::permit<NonInternalType>();
+  }
+
+  // success: the base type is internal
+  public fun t3() {
+    internal::permit<InternalGeneric<NonInternalType>>();
+  }
+
+  // fail: the base type is not internal
+  public fun t4() {
+    internal::permit<Permit<Internal>>();
+  }
+}
+
+module a::other {
+  struct NonInternalType {}
+}
+
+module std::internal {
+  struct Permit<phantom T> has drop {}
+
+  public fun permit<T>(): Permit<T> { Permit {} }
+}
```

Comment:
nit, can you fix the indent width (4 instead of 2)

Reply:
I made it same style as everywhere else in these tests. I'll run a formatter on all of them in a follow up.